addons/mail/static/src/components/attachment_viewer/attachment_viewer.js

```diff
@@ -69,6 +70,12 @@ class AttachmentViewer extends Component {
         document.addEventListener('click', this._onClickGlobal);
     }
 
+    willPatch() {
+        if (this._imageRef.el) {
+            this._imageRef.el.removeEventListener('load', this._onLoadImage);
+        }
+    }
```

Comment:
This one should have a similar condition to the one in `_handleImageLoad`

`this._renderedAttachment !== this.attachmentViewer.attachment`
Because otherwise will might remove the listener but never add it back.

Reply:
Alternative is to always remove and then re-add the listener, so we don't need to worry about such conditions. Not sure why we didn't in the first place, maybe it looked like an optimization?

cc @alexkuhn

Reply:
> Alternative is to always remove and then re-add the listener, so we don't need to worry about such conditions. Not sure why we didn't in the first place, maybe it looked like an optimization?
> 
> cc @alexkuhn

Fine by me as fix. `_handleImageLoad` is clumsy code-reuse, but that's something to be cleaned in master.

Reply:
@seb-odoo j'ai mis a jour la branche, j'ai aussi rajouté un test et j'ai vérifié qu'il ne passait pas sans le fix.